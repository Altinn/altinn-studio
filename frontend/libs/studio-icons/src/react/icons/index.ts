export { AccordionIcon } from './AccordionIcon';
export { ArrayIcon } from './ArrayIcon';
export { BooleanIcon } from './BooleanIcon';
export { CheckboxIcon } from './CheckboxIcon';
export { CombinationIcon } from './CombinationIcon';
export { ConfirmationTaskIcon } from './ConfirmationTaskIcon';
export { DataTaskIcon } from './DataTaskIcon';
export { DefinitionIcon } from './DefinitionIcon';
export { ElementIcon } from './ElementIcon';
export { EndEventIcon } from './EndEventIcon';
export { FeedbackTaskIcon } from './FeedbackTaskIcon';
export { GiteaIcon } from './GiteaIcon';
export { GitHubIcon } from './GitHubIcon';
export { GroupIcon } from './GroupIcon';
export { LikertIcon } from './LikertIcon';
export { LongTextIcon } from './LongTextIcon';
export { NavBarIcon } from './NavBarIcon';
export { NumberIcon } from './NumberIcon';
export { ObjectIcon } from './ObjectIcon';
export { TextIcon } from './TextIcon';
export { PaymentDetailsIcon } from './PaymentDetailsIcon';
export { PaymentTaskIcon } from './PaymentTaskIcon';
export { PropertyIcon } from './PropertyIcon';
export { RadioButtonIcon } from './RadioButtonIcon';
export { ReferenceIcon } from './ReferenceIcon';
export { RepeatingGroupIcon } from './RepeatingGroupIcon';
export { SectionHeaderWarningIcon } from './SectionHeaderWarningIcon';
export { SelectIcon } from './SelectIcon';
export { ShortTextIcon } from './ShortTextIcon';
export { SignTaskIcon } from './SignTaskIcon';
export { SlackIcon } from './SlackIcon';
export { StringIcon } from './StringIcon';
export { TitleIcon } from './TitleIcon';
