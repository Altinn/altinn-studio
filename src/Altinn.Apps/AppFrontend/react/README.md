## Running the tests

### Lint checks

1. Navigate to the folder `src/Altinn.Apps/AppFrontend/react`.
2. Execute `npm ci`. This step is only nescessary if you have not already done it, or if you change branches.
3. Execute `npm run install-deps`. This step is only nescessary if you have not already done it, or if you change branches.
4. Execute `npm run lint`.

### Unit tests

1. Navigate to the folder `src/Altinn.Apps/AppFrontend/react`.
2. Execute `npm ci`. This step is only nescessary if you have not already done it, or if you change branches.
3. Execute `npm run install-deps`. This step is only nescessary if you have not already done it, or if you change branches.
4. Execute `npm run test`.
