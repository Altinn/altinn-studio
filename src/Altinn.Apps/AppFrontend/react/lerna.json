{
  "command": {
    "bootstrap": {
      "hoist": true
    }
  },
  "packages": [
    "./*"
  ],
  "version": "independent"
}
