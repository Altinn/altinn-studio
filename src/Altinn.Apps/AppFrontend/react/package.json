{
	"name": "app-frontend",
	"private": true,
	"scripts": {
		"install-deps": "lerna bootstrap --hoist --ci",
		"no-ci-hoist": "lerna bootstrap --hoist --no-ci",
		"install-deps-with-lock": "lerna bootstrap",
		"test": "lerna exec --concurrency 1 -- npm test",
		"install-all": "lerna exec -- npm ci",
		"clean-lock": "lerna exec -- rm ./package-lock.json",
		"clean-node-modules": "rm -rf ./node_modules && lerna exec --concurrency 1 -- rm -rf ./node_modules",
		"update-packages": "lerna exec --concurrency 1 -- npm update --dev",
		"list-outdated": "lerna run list-outdated"
	},
	"devDependencies": {
		"@typescript-eslint/eslint-plugin": "~4.8.2",
		"@typescript-eslint/parser": "~4.8.2",
		"eslint": "~7.14.0",
		"eslint-config-airbnb": "~18.2.1",
		"eslint-plugin-import": "~2.22.1",
		"eslint-plugin-jsx-a11y": "~6.4.1",
		"eslint-plugin-react": "~7.21.5",
		"eslint-plugin-react-hooks": "^4.2.0",
		"ini": "1.3.5",
		"lerna": "3.22.1"
	}
}
