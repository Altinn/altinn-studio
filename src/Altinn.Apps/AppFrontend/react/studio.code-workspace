{
	"folders": [
		{
			"path": "."
		}
  ],
  "settings": {
    "typescript.surveys.enabled": false,
    "typescript.tsdk": "node_modules/typescript/lib",
    "eslint.alwaysShowStatus": true,
    "eslint.validate": [
      "javascript",
      "javascriptreact",
      { "language": "typescript", "autoFix": true },
      { "language": "typescriptreact", "autoFix": true}
    ]
  }
}
