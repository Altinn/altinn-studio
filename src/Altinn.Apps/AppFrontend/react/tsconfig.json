{
  "compilerOptions": {
    "outDir": "compiled",
    "module": "commonjs",
    "target": "es5",
    "lib": [
      "es6",
      "dom",
      "es7",
      "esnext"
    ],
    "sourceMap": true,
    "allowJs": true,
    "jsx": "react",
    "moduleResolution": "node",
    "rootDir": "./",
    "noImplicitReturns": true,
    "noImplicitThis": true,
    "noImplicitAny": true,
    "strictNullChecks": false,
    "noUnusedLocals": true
  },
  "include": [
    "**/*.ts",
    "**/*.tsx",
  ],
  "exclude": [
    "node_modules",
    "build",
    "scripts",
    "acceptance-tests",
    "webpack",
    "jest",
    "setupTests.ts",
    "templates",
    "**/setupTests.ts"
  ],
}
