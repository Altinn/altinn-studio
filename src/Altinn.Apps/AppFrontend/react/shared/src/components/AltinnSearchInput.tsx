import { createTheme, FormControl, InputAdornment, TextField } from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';
import classNames from 'classnames';
import * as React from 'react';
import altinnTheme from '../theme/altinnStudioTheme';

export interface IAltinnSearchInputComponentProvidedProps {
  classes: any;
  id: string;
  placeholder: any;
  onChangeFunction: any;
  ariaLabel: string;
}

const theme = createTheme(altinnTheme);

const styles = {
  searchBox: {
    border: '1px solid ' + theme.altinnPalette.primary.blueDark,
    marginTop: '10px',
    marginBottom: '24px',
    background: 'none',
    width: '386px',
  },
  searchBoxInput: {
    fontSize: '16px',
    color: '#000000',
    padding: '6px',
  },
  searchBoxIcon: {
    color: '#000000',
    fontSize: '30px',
    marginRight: '10px',
  },
};


export class AltinnSearchInput extends React.Component<IAltinnSearchInputComponentProvidedProps> {
  public render() {
    const { classes } = this.props;
    return (
      <FormControl
        classes={{ root: classNames(classes.searchBox) }}
        fullWidth={true}
      >
        <TextField
          id={this.props.id}
          placeholder={this.props.placeholder}
          onChange={this.props.onChangeFunction}
          inputProps={{
            "aria-label": this.props.ariaLabel,
          }}
          InputProps={{
            disableUnderline: true,
            startAdornment:
              <InputAdornment
                position={'end'}
                classes={{ root: classNames(classes.searchBoxIcon) }}
              >
                <i
                  className={'fa fa-search'}
                  title={'Søkeikon'}
                  aria-label={'Søkeikon'}
                />
              </InputAdornment>,
            classes: { root: classNames(classes.searchBoxInput) },
          }}
        />
      </FormControl>
    );
  }
}

export default withStyles(styles)(AltinnSearchInput);
