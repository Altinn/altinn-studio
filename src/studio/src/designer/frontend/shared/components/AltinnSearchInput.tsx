import { createMuiTheme, FormControl, InputAdornment, TextField } from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';
import classNames from 'classnames';
import * as React from 'react';
import altinnTheme from '../theme/altinnStudioTheme';

export interface IAltinnSearchInputComponentProvidedProps {
  classes: any;
  id: string;
  placeholder: any;
  onChangeFunction: any;
  ariaLabel: string;
}

export interface IAltinnSearchInputComponentState {
}

const theme = createMuiTheme(altinnTheme);

const styles = {
  searchBox: {
    border: '1px solid ' + theme.altinnPalette.primary.blueDark,
    marginTop: '10px',
    marginBottom: '24px',
    background: 'none',
    width: '386px',
  },
  searchBoxInput: {
    fontSize: '16px',
    color: '#000000',
    padding: '6px',
  },
  searchBoxIcon: {
    color: '#000000',
    fontSize: '30px',
    marginRight: '10px',
  },
};

// eslint-disable-next-line max-len
export class AltinnSearchInput extends React.Component<IAltinnSearchInputComponentProvidedProps, IAltinnSearchInputComponentState> {
  public render() {
    const { classes } = this.props;
    return (
      <FormControl
        classes={{ root: classNames(classes.searchBox) }}
        fullWidth={true}
      >
        <TextField
          id={this.props.id}
          placeholder={this.props.placeholder}
          onChange={this.props.onChangeFunction}
          inputProps={{
            "aria-label": this.props.ariaLabel,
          }}
          InputProps={{
            disableUnderline: true,
            startAdornment:
              <InputAdornment
                position={'end'}
                classes={{ root: classNames(classes.searchBoxIcon) }}
              >
                <i
                  className={'fa fa-search'}
                  title={'Søkeikon'}
                  aria-label={'Søkeikon'}
                />
              </InputAdornment>,
            classes: { root: classNames(classes.searchBoxInput) },
          }}
        />
      </FormControl>
    );
  }
}

export default withStyles(styles)(AltinnSearchInput);
