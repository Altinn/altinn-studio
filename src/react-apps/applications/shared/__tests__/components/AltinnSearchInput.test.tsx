import 'jest';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import AltinnSearchInput from '../../src/components/AltinnSearchInput';

describe('>>> AltinnSearchInput--- Snapshot', () => {
  let mockId: string;
  let mockPlaceholder: any;
  let mockOnChangeFunction: any;
  beforeEach(() => {
    mockId = 'mock id';
    mockPlaceholder = 'mock placeholder';
    mockOnChangeFunction = () => {
      return false;
    };
  });

  it('+++ Should match snapshot with the least amount of params', () => {
    const rendered = renderer.create(
      <AltinnSearchInput
        id={mockId}
        placeholder={mockPlaceholder}
        onChangeFunction={mockOnChangeFunction}
      />,
    );
    expect(rendered).toMatchSnapshot();
  });
});
