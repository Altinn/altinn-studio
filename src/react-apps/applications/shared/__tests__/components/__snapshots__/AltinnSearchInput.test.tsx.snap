// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> AltinnSearchInput--- Snapshot +++ Should match snapshot with the least amount of params 1`] = `
<div
  className="MuiFormControl-root AltinnSearchInput-searchBox-1 MuiFormControl-fullWidth"
>
  <div
    className="MuiFormControl-root MuiTextField-root"
  >
    <div
      className="MuiInputBase-root MuiInput-root AltinnSearchInput-searchBoxInput-2 MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart"
      onClick={[Function]}
    >
      <div
        className="MuiInputAdornment-root AltinnSearchInput-searchBoxIcon-3 MuiInputAdornment-positionEnd"
      >
        <i
          className="fa fa-search"
        />
      </div>
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart"
        disabled={false}
        id="mock id"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="mock placeholder"
        required={false}
        type="text"
      />
    </div>
  </div>
</div>
`;
